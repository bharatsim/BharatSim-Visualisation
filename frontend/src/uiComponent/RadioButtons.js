import React from 'react';

import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormControl from '@material-ui/core/FormControl';
import RadioGroup from '@material-ui/core/RadioGroup';
import { useController } from 'react-hook-form';
import Radio from '@material-ui/core/Radio';
import PropTypes from 'prop-types';
import { Box } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
  vertical: {
    display: 'flex',
    flexDirection: 'column',
  },
  horizontal: {
    display: 'flex',
    flexDirection: 'row',
  },
}));

function RadioButtons({ options, control, name, defaultValue, vertical, disabled }) {
  const classes = useStyles();
  const {
    field: { onChange, value },
  } = useController({
    name,
    control,
    defaultValue,
  });
  return (
    <FormControl disabled={disabled}>
      <RadioGroup value={value} onChange={onChange}>
        <Box className={vertical ? classes.vertical : classes.horizontal}>
          {options.map((option) => {
            return (
              <FormControlLabel
                value={option.value}
                control={<Radio size="small" />}
                label={option.label}
                key={`option-${option.value}`}
                name={option.label}
              />
            );
          })}
        </Box>
      </RadioGroup>
    </FormControl>
  );
}

RadioButtons.defaultProps = {
  vertical: true,
  disabled: false,
};

RadioButtons.propTypes = {
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    }),
  ).isRequired,
  control: PropTypes.shape({}).isRequired,
  name: PropTypes.string.isRequired,
  defaultValue: PropTypes.string.isRequired,
  vertical: PropTypes.bool,
  disabled: PropTypes.bool,
};

export default RadioButtons;
