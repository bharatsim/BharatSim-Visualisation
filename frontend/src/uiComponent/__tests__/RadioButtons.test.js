import React from 'react';
import { render } from '@testing-library/react';
import { useForm } from 'react-hook-form';
import RadioButtons from '../RadioButtons';

const TestForm = ({ onSubmit, vertical }) => {
  const { control, handleSubmit } = useForm({ mode: 'onChange' });
  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <RadioButtons
        options={[
          { value: 'option1', label: 'option1' },
          { value: 'option2', label: 'option2' },
        ]}
        control={control}
        defaultValue="option1"
        name="radioButtons"
        vertical={vertical}
      />
      <button type="submit">submit</button>
    </form>
  );
};

describe('<RadioButtons />', () => {
  it('should match snapshot vertical direction', () => {
    const { container } = render(<TestForm onSubmit={jest.fn()} vertical />);

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot horizontal direction', () => {
    const { container } = render(<TestForm onSubmit={jest.fn()} vertical={false} />);

    expect(container).toMatchSnapshot();
  });
});
