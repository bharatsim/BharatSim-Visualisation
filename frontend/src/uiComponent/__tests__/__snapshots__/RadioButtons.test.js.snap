// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButtons /> should match snapshot horizontal direction 1`] = `
<div>
  <form>
    <div
      class="MuiFormControl-root"
    >
      <div
        class="MuiFormGroup-root"
        role="radiogroup"
      >
        <div
          class="MuiBox-root MuiBox-root-13 makeStyles-horizontal-12"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-15 Mui-checked MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input-17"
                  name="option1"
                  type="radio"
                  value="option1"
                />
                <div
                  class="PrivateRadioButtonIcon-root-18 PrivateRadioButtonIcon-checked-20"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-19 MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              option1
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-17"
                  name="option2"
                  type="radio"
                  value="option2"
                />
                <div
                  class="PrivateRadioButtonIcon-root-18"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-19 MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              option2
            </span>
          </label>
        </div>
      </div>
    </div>
    <button
      type="submit"
    >
      submit
    </button>
  </form>
</div>
`;

exports[`<RadioButtons /> should match snapshot vertical direction 1`] = `
<div>
  <form>
    <div
      class="MuiFormControl-root"
    >
      <div
        class="MuiFormGroup-root"
        role="radiogroup"
      >
        <div
          class="MuiBox-root MuiBox-root-3 makeStyles-vertical-1"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-5 Mui-checked MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input-7"
                  name="option1"
                  type="radio"
                  value="option1"
                />
                <div
                  class="PrivateRadioButtonIcon-root-8 PrivateRadioButtonIcon-checked-10"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-9 MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              option1
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-7"
                  name="option2"
                  type="radio"
                  value="option2"
                />
                <div
                  class="PrivateRadioButtonIcon-root-8"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-9 MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              option2
            </span>
          </label>
        </div>
      </div>
    </div>
    <button
      type="submit"
    >
      submit
    </button>
  </form>
</div>
`;
